contracts/solve/src/SolverNetInbox.sol

```diff
@@ -203,37 +183,36 @@ contract SolverNetInbox is OwnableRoles, ReentrancyGuard, Initializable, Deploye
     }
 
     /**
-     * @notice Cancel an open and refund deposits.
-     * @dev Only order initiator can cancel.
+     * @notice Close order and refund deposits after fill deadline has elapsed.
+     * @dev Only order initiator can close.
      * @param id ID of the order.
      */
-    function cancel(bytes32 id) external nonReentrant {
+    function close(bytes32 id) external nonReentrant {
         OrderState memory state = _orderState[id];
-        address user = _orderHeader[id].owner;
+        SolverNet.Header memory header = _orderHeader[id];
 
         if (state.status != Status.Pending) revert OrderNotPending();
-        if (user != msg.sender) revert Unauthorized();
+        if (header.owner != msg.sender) revert Unauthorized();
+        if (header.fillDeadline >= block.timestamp) revert OrderStillValid();
```

Comment:
Add buffer 🤝 

Also some other maybe ideas for error name`OrderNotExpired()` ,  `OrderStillOpen()`? no hard preferences just spit balling